roles/publish-charm/tasks/main.yaml

```diff
@@ -0,0 +1,52 @@
+- name: Publish charms to charmhub
+  when: publish_charm
+  environment:
+    CHARMCRAFT_AUTH: "{{ charmhub_token.value }}"
+  block:
+    - name: Upload oci-image to charmhub
+      register: upload_oci_image_output
+      vars:
+        metadata: "{{ lookup('file', zuul.project.src_dir+'/metadata.yaml') | from_yaml }}"
+      args:
+        executable: /bin/bash
+      shell: |
+        set -x
+        image={{ item.value['upstream-source'] }}
+        docker pull $image
```

Comment:
Will this fail somewhat frequently with the API limitations on docker hub for unauthenticated users? Especially since all pulls will come from the same set of IPs?

I know that this is dependent upon a variety of factors, but do we need an image registry where we source these files from? This would probably also require a separate task to configure the local docker registry to point to this registry?

Reply:
Pushed a new patch which will check for docker mirror registry configuration and pulls image from mirror registry.